plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftShutdownController.java

```diff
@@ -36,9 +41,23 @@ public OpenShiftServerBehaviour getBehavior() {
 	@Override
 	public void stop(boolean force) {
 		OpenShiftServerBehaviour behavior = getBehavior();
+		IServer server = behavior.getServer();
+		IDeploymentConfig dc = OpenShiftServerUtils.getDeploymentConfig(server);
+		boolean debugMode = OpenShiftLaunchController.DEBUG_MODE.equals(server.getMode());
+
 		behavior.setServerStopping();
+
 		try {
-			OpenShiftDebugUtils.get().terminateRemoteDebugger(behavior.getServer());
+			if(dc == null) {
+				//something is wrong, do as much as possible.
+				OpenShiftDebugUtils.get().terminateRemoteDebugger(behavior.getServer());
+			} else if(debugMode) {
+				OpenShiftLaunchController c =
+//						(OpenShiftLaunchController)behavior.getController("launch");
```

Comment:
I'd rather prefer we call `(OpenShiftLaunchController)behavior.getController("launch");`

worst case scenario, it's exactly the same as creating a new instance, best case scenario, if the behavior changes in the implementation, we get to reuse an existing instance. Either way it's transparent from our PoV